airbyte-config/models/src/main/java/io/airbyte/config/EnvConfigs.java

```diff
@@ -438,8 +441,29 @@ private TolerationPOJO parseToleration(final String tolerationStr) {
    */
   @Override
   public Map<String, String> getJobKubeNodeSelectors() {
+    return getNodeSelectorsFromEnvString(getEnvOrDefault(JOB_KUBE_NODE_SELECTORS, ""));
+  }
+
+  /**
+   * Returns a map of node selectors from its own environment variable for the Check job specifically.
+   * Falls back on JOB_KUBE_NODE_SELECTORS if CHECK_JOB_KUBE_NODE_SELECTORS is not set. The value of
+   * the env is a string that represents one or more node selector labels. Each kv-pair is separated
+   * by a `,`
+   * <p>
+   * For example:- The following represents two node selectors
+   * <p>
+   * airbyte=server,type=preemptive
+   *
+   * @return map containing kv pairs of node selectors
+   */
+  @Override
+  public Map<String, String> getCheckJobKubeNodeSelectors() {
```

Comment:
would it make sense for this to be `Optional<Map<String, String>>` to emphasize this configuration is optional? My current reading of the code is that we will always create a map even if this env var is unset. If it is unset it'll just be an empty map. Codifying Optional.empty instead of empty map would be a bit clearer to me.

Reply:
maybe for `getJobKubeNodeSelectors()`  too?

Reply:
Done!